### PR TITLE
Changes over credits info about providers

### LINKED ISSUE #9190: Credits info changes
- [x] Hide credits info when non-free analysis is running or parameters have not changed. Just show `Apply` button as desactivated (cc @piensaenpixel @saleiva ).
- [x] Distinguish providers for non-free analysis (cc @ethervoid). For now, geocoding, AOI and routing. Values: `heremaps`, `mapzen` or `google`. Take into account organization attributes prevail over user attributes.

### PR DESCRIPTION
**Pull request by xavijam**
UI changes, if analysis is running:
screen shot 2016-08-02 at 11 29 35

If analysis is done and form is valid:
screen shot 2016-08-02 at 11 29 43

If analysis is done but form is invalid:
screen shot 2016-08-02 at 11 31 06

- [x] Adding georeference by street address like non-free analysis + checking providers for different purposes (cc: @ethervoid).
- [x] UI changes:
  - Don't display APPLY button enabled when form doesn't have any change.
  - Don't show credits info when analysis is running or there is no changes in the form.

Fixes #9190 
CR: @nobuti

### CODE REVIEW

**Comment by ethervoid** [comment]
Looks pretty good mate :)

**Comment by xavijam** [comment]
UI changes, if analysis is running:
screen shot 2016-08-02 at 11 29 35

If analysis is done and form is valid:
screen shot 2016-08-02 at 11 29 43

If analysis is done but form is invalid:
screen shot 2016-08-02 at 11 31 06

cc @saleiva @piensaenpixel

**Comment by saleiva** [comment]
Quick remember about adding the MAX credits on the yellow area. Text would be something like

"This will take XXXmax credits from your account. Extra costs will be charged at..."

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analyses-quota-factory.js [hunk 1]
```diff
@@ -1,37 +1,87 @@
...
+var QUOTA_DEFAULTS = {
+  totalQuota: 0,
+  usedQuota: 0,
+  blockSize: 1000,
+  blockPrice: 0
+};
+
+function parseQuota (data) {
+  // Remove null | undefined | falsy values
+  data = _.pick(data, _.identity);
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analyses-quota-factory.js]: Also `data.filter(Boolean);` would do the trick.

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analyses-quota-factory.js [hunk 2]
```diff
@@ -1,37 +1,87 @@
...
+      blockPrice: data.block_price,
+      hardLimit: data.hard_limit !== undefined ? data.hard_limit : true
+    },
+    QUOTA_DEFAULTS
+  );
+}
+
+function getProviderTypeByUser (type, userModel) {
+  if (userModel.isInsideOrg()) {
+    var organizationModel = userModel._organizationModel;
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analyses-quota-factory.js]: Just a matter of style, but try to declare vars at the start of the function scope.

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analyses-quota-factory.js]: But `organizationModel` should be undefined if user is not inside an org.

Comment (nobuti) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analyses-quota-factory.js]: Better `userModel. getOrganization()`

Comment (nobuti) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analyses-quota-factory.js]: I meant something like this:
```
var organizationModel;
if (userModel.isInsideOrg()) {
     organizationModel = wadus;
}
```

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analyses-quota-factory.js [hunk 3]
```diff
@@ -1,37 +1,87 @@
...
+    }
+    return false;
   },
 
-  hasQuota: function (type) {
-    var quotaInfo = ANALYSES_WITH_QUOTA_MAP[type];
-    return quotaInfo.totalQuota > quotaInfo.usedQuota;
+  hasQuota: function (type, userModel) {
+    if (ANALYSES_WITH_QUOTA_MAP[type]) {
+      var quotaInfo = ANALYSES_WITH_QUOTA_MAP[type](userModel);
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analyses-quota-factory.js]: same as above

**Comment by nobuti** [comment]
👍

**Comment by xavijam** [comment]
Feedback applied

**Comment by xavijam** [comment]
Frontend tests passing (finally!), merging!

**Pull request merged**